cmd/k8sagent/unit/agent.go

```diff
@@ -0,0 +1,270 @@
+// Copyright 2012-2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package unit
+
+import (
+	"runtime"
+	"time"
+
+	"github.com/juju/clock"
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/featureflag"
+	"github.com/juju/gnuflag"
+	"github.com/juju/loggo"
+	"github.com/juju/names/v4"
+	"github.com/juju/utils/voyeur"
+	"github.com/juju/worker/v2"
+	"github.com/juju/worker/v2/dependency"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/api/uniter"
+	jujucmd "github.com/juju/juju/cmd"
+	jujudagent "github.com/juju/juju/cmd/jujud/agent"
+	cmdutil "github.com/juju/juju/cmd/jujud/util"
+	"github.com/juju/juju/core/machinelock"
+	"github.com/juju/juju/upgrades"
+	jujuversion "github.com/juju/juju/version"
+	jworker "github.com/juju/juju/worker"
+	"github.com/juju/juju/worker/gate"
+	"github.com/juju/juju/worker/introspection"
+	"github.com/juju/juju/worker/logsender"
+	"github.com/juju/juju/worker/upgradesteps"
```

Comment:
I don't think we are going to need upgradesteps worker, because we will be recreating the container on upgrades, so no need to have anything running upgrade steps.

Reply:
ah, right!